editor/editor_storage.hpp

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include "std/unique_ptr.hpp"
+
+namespace pugi
+{
+class xml_document;
+}
+
+namespace editor
+{
+// Editor storage interface.
+class StorageBase
+{
+public:
+  virtual ~StorageBase() = default;
+  virtual bool Save(pugi::xml_document const & doc) = 0;
+  virtual bool Load(pugi::xml_document & doc) = 0;
+  virtual void Reset() = 0;
+};
+
+// Class which saves/loads edits to/from local file.
+class StorageLocal : public StorageBase
+{
+public:
+  bool Save(pugi::xml_document const & doc) override;
+  bool Load(pugi::xml_document & doc) override;
+  void Reset() override;
+};
+
+// Class which save/load edits to/from xml_document class instance.
+class StorageMemory : public StorageBase
+{
+public:
+  StorageMemory();
+  bool Save(pugi::xml_document const & doc) override;
```

Comment:
Why? We have only one base class and every method is marked as override explicitly.